ckanext/announcements/tests/fixtures.py

```diff
@@ -1,27 +1,36 @@
 import contextlib
 import pytest
 
-from ckan.cli.db import _resolve_alembic_config
-import ckan.model as model
+from ckan.plugins import toolkit
 
 
-# TODO: in the next CKAN release
-# there will be a more elegant way to achieve this
-@contextlib.contextmanager
-def _repo_for_plugin(plugin):
-    original = model.repo._alembic_ini
-    model.repo._alembic_ini = _resolve_alembic_config(plugin)
-    try:
-        yield model.repo
-    finally:
-        model.repo._alembic_ini = original
+if toolkit.check_ckan_version(max_version='2.11'):
 
+    from ckan.cli.db import _resolve_alembic_config
+    import ckan.model as model
 
-def _apply_alembic_migrations():
-    with _repo_for_plugin("announcements") as repo:
-        repo.upgrade_db("head")
+    # TODO: in the next CKAN release
+    # there will be a more elegant way to achieve this
+    @contextlib.contextmanager
+    def _repo_for_plugin(plugin):
+        original = model.repo._alembic_ini
+        model.repo._alembic_ini = _resolve_alembic_config(plugin)
+        try:
+            yield model.repo
+        finally:
+            model.repo._alembic_ini = original
 
+    def _apply_alembic_migrations():
+        with _repo_for_plugin("announcements") as repo:
+            repo.upgrade_db("head")
 
-@pytest.fixture
-def announcement_migrate():
-    _apply_alembic_migrations()
+    @pytest.fixture
+    def announcement_migrate():
+
+        _apply_alembic_migrations()
```

Comment:
@avdata99 I'm not exactly sure what's going on here. Is the `migrate_db_for` fixture broken in CKAN 2.11?

Reply:
I can't manage to find the problem @pdelboca 
I added the `with_plugins` fixture and now it's fixed

Reply:
@avdata99 I suggest to change this fixture for the following one:

```python
import pytest


@pytest.fixture
def clean_db(reset_db, migrate_db_for):
    reset_db()
    migrate_db_for("announcements")
```

The with_plugins is necessary to force test to use the `ckan.plugins` definition of the `test.ini` file.  Without the fixture, it will be ignored.

Unit testing with CKAN is in my opinion confusing. There is a [whole discussion about it](https://github.com/ckan/ckan/discussions/6373), while it didn't got anywhere I think it documents what's going on when testing CKAN. I will definitely recommend making a 🧉 and read it.